File: Section1 - Basics/Patterns/pattern22.js
function printSquare(n) {
	let str = "";
	// Print the upper half of the pattern
	for (let i = n; i >= 1; i--) {
		for (let j = n; j >= 1; j--) {
			str = str + Math.max(i, j) + " ";
		}
		for (let j = 2; j <= n; j++) {
			str = str + Math.max(i, j) + " ";
		}
		str += "\n";
	}

	// Print the lower half of the pattern
	for (let i = 2; i <= n; i++) {
		for (let j = n; j >= 1; j--) {
			str = str + Math.max(i, j) + " ";
		}
		for (let j = 2; j <= n; j++) {
			str = str + Math.max(i, j) + " ";
		}
		str += "\n";
	}
	console.log(str);
}

printSquare(4);

File: Section1 - Basics/Patterns/pattern22.java
void printSquare(int n) {
   // Print the upper half of the pattern
   for (int i = n; i >= 1; i--) {
       for (int j = n; j >= 1; j--) {
           System.out.print(Math.max(i, j) + " ");
       }
       for (int j = 2; j <= n; j++) {
           System.out.print(Math.max(i, j) + " ");
       }
       System.out.println();
   }
   
   // Print the lower half of the pattern
   for (int i = 2; i <= n; i++) {
       for (int j = n; j >= 1; j--) {
           System.out.print(Math.max(i, j) + " ");
       }
       for (int j = 2; j <= n; j++) {
           System.out.print(Math.max(i, j) + " ");
       }
       System.out.println();
   }
}